### PR TITLE
UO.TryGetProperty should determine the result value

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue https://github.com/microsoft/Power-Fx/issues/2591.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs [hunk 1]
```diff
@@ -557,21 +557,10 @@ private static FormulaValue OperatorDynamicGetField(BinaryOpNode node, FormulaVa
...
-                        return new UntypedObjectValue(node.IRContext, res);
-                    }
-                    else
-                    {
-                        return new BlankValue(node.IRContext);
-                    }
+                if (cov.Impl is UntypedObjectBase untypedObjectBase && cov.Impl.Type is ExternalType et && (et.Kind == ExternalTypeKind.Object || et.Kind == ExternalTypeKind.ArrayAndObject))
+                {
+                    // Passing Span in case host wants to return an error.
+                    return untypedObjectBase.GetProperty(sv.Value, node.IRContext.ResultType, node.IRContext.SourceContext);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs:563]: >node.IRContext.SourceContext [](http://example.com/codeflow?start=94&length=28)

I don't think we have any meaningful span here. 
(we should just remove IRContext.SourceContext) 

I'd just remove this param.

File: src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs [hunk 2]
```diff
@@ -557,21 +557,10 @@ private static FormulaValue OperatorDynamicGetField(BinaryOpNode node, FormulaVa
...
-                            return new BlankValue(node.IRContext);
-                        }
-
-                        return new UntypedObjectValue(node.IRContext, res);
-                    }
-                    else
-                    {
-                        return new BlankValue(node.IRContext);
-                    }
+                if (cov.Impl is UntypedObjectBase untypedObjectBase && cov.Impl.Type is ExternalType et && (et.Kind == ExternalTypeKind.Object || et.Kind == ExternalTypeKind.ArrayAndObject))
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs:560]: >UntypedObjectBase [](http://example.com/codeflow?start=32&length=17)

What if we get something that just implements the interface?

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Public/Values/UntypedObjectValue.cs [hunk 3]
```diff
@@ -7,6 +7,7 @@
 using System.Diagnostics.Contracts;
 using System.Text;
 using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Syntax;
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/Values/UntypedObjectValue.cs:10]: nit, remove?

**Review by MikeStall — DISMISSED** [review]
:shipit:

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**